src/app/doctor-login/doctor-login.component.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { Doctor } from '../Doctor';
+import { DoctorService } from '../doctor.service';
+import { MatDialogRef } from '@angular/material';
+import { MatDialog } from '@angular/material';
+import { Router } from '@angular/router';
+import { DoctorTokenStorage} from '../doctorTokenStorage';
+import { AlertsService } from 'angular-alert-module';  
+import {MatSnackBar} from '@angular/material';
+import { InvalidDoctorComponent } from './invalid-doctor.component';
+import { FailRegisterComponent } from './fail-register.component';
+
+@Component({
+  selector: 'app-doctor-login',
+  templateUrl: './doctor-login.component.html',
+  styleUrls: ['./doctor-login.component.css']
+})
+export class DoctorLoginComponent implements OnInit {
+
+  doctor;
```

Comment:
sir,doctor object is for login and doctor1 object is for registering.since two way data binding is used,same object cant be used for both login and registration.